### PR TITLE
Parameterized font errors

### LINKED ISSUE #201: Parameterise font name in Fontist::Errors::NonSupportedFontError
When a font being installed by a metanorma flavour is not supported, metanorma-standoc prints its own error message:

https://github.com/metanorma/metanorma-standoc/blob/046b4ed4cf6b767a625c818cb4c0cd1646cce3f0/lib/asciidoctor/standoc/converter.rb#L142

````
rescue Fontist::Errors::NonSupportedFontError => e
          font = /Font '([^']+)'/.match(e.to_s)[1]
          Metanorma::Util.log("[fontist] '#{font}' font is not supported. " \
            "Please go to github.com/metanorma/metanorma-#{flavor_name}/issues" \
            " to report this issue.", :info)
````

I've had to insert the ugly `font = /Font '([^']+)'/.match(e.to_s)[1]`, because the original developer didn't define `font` at all.

It would of course be far preferable if the raised error could give the font name as a parameter, as in https://stackoverflow.com/questions/11636874/raise-custom-exception-with-arguments, so that I can instead just use `e.font`.

### PR DESCRIPTION
**Pull request by alexeymorozov**
Closes #201

### CODE REVIEW

File: lib/fontist/errors.rb [hunk 1]
```diff
@@ -9,10 +10,26 @@ class LicensingError < GeneralError; end
...
+
+      def initialize(msg, font = nil, style = nil)
+        @font = font
+        @style = style
+
+        super(msg)
+      end
+
+      def name
+        "#{@font} #{@style}"
```
Comment (ronaldtse) [inline lib/fontist/errors.rb:28]: The "font" and "style" names will be confusing if they contain spaces, please separate them with a pattern that is notable, "Font name: 'xxx'; Style: 'xxx'"

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: spec/fontist/manifest/install_spec.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@
       it "raises non-supported font error" do
         no_fonts do
           expect { command }.to raise_error Fontist::Errors::NonSupportedFontError
+          expect { command }.to(raise_error { |e| expect(e.font).to eq "Unexisting Font" })
```
Comment (ronaldtse) [inline spec/fontist/manifest/install_spec.rb:24]: "Non-existing font"

File: spec/fontist/font_spec.rb [hunk 3]
```diff
@@ -488,6 +496,8 @@
 
       it "raises font unsupported error" do
         expect { command }.to raise_error Fontist::Errors::NonSupportedFontError
+
+        expect { command }.to(raise_error { |e| expect(e.font).to eq "unexisting" })
```
Comment (ronaldtse) [inline spec/fontist/font_spec.rb:500]: "nonexistent"

File: spec/fontist/font_spec.rb [hunk 4]
```diff
@@ -411,6 +416,8 @@
...
```
Comment (ronaldtse) [inline spec/fontist/font_spec.rb:420]: "nonexistent"

File: lib/fontist/font.rb [hunk 5]
```diff
@@ -96,11 +96,11 @@ def formula
 
     def downloadable_font
       if formula
-        raise(
-          Fontist::Errors::MissingFontError,
+        raise Fontist::Errors::MissingFontError.new(
```
Comment (ronaldtse) [inline lib/fontist/font.rb:99]: Should just pass the "name" to the error and have the Error class build the message.

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/fontist/errors.rb (hunk 1 above)
Comment (alexeymorozov) [reply lib/fontist/errors.rb]: Done

File: spec/fontist/manifest/install_spec.rb (hunk 2 above)
Comment (alexeymorozov) [reply spec/fontist/manifest/install_spec.rb]: Thanks for also improving my English :)

File: lib/fontist/font.rb (hunk 5 above)
Comment (alexeymorozov) [reply lib/fontist/font.rb]: Now it's much easier to throw these errors :)

File: lib/fontist/errors.rb [hunk 6]
```diff
@@ -9,10 +10,56 @@ class LicensingError < GeneralError; end
...
+
+      def prepare_name(font, style)
+        names = []
+        names << "'#{font}'"
+        names << "'#{style}'" if style
+        names.join(" ")
+      end
+    end
+
+    class NonSupportedFontError < FontError
```
Comment (ronaldtse) [inline lib/fontist/errors.rb:53]: Can we rename this to be the `UnsupportedFontError`?

**Review by ronaldtse — CHANGES_REQUESTED** [review]
The rest are good. Thanks!

File: lib/fontist/errors.rb (hunk 6 above)
Comment (alexeymorozov) [reply lib/fontist/errors.rb]: Of course! Let's correct all these `non` and `un` prefixes.

**Comment by ronaldtse** [comment]
LGTM, thanks @alexeymorozov !

**Pull request merged**